### PR TITLE
Landau and Vavilov energy loss

### LINKED ISSUE #174: Add Landau and Vavilov energy loss PDF

### PR DESCRIPTION
**Pull request by grzanka**
Fixes #174

### CODE REVIEW

**Comment by grzanka** [comment]
@marwin1991 funny - preview on web_dev doesn't work, but compiling it locally or when using `*wasm` files from *web* repo everythin works well

**Comment by grzanka** [comment]
@marwin1991 now it looks ok in WebDev ! any other comments ?

**Comment by marwin1991** [comment]
For  default values and Vavilov we have enrgy
Energy[Mev] :  386.77354709418773  PDF [a.u.] : 0
Energy[Mev] :  388.7775551102198    PDF [a.u.] : 0.020241360601070967
.
.
.
Energy[Mev]: 959.9198396793696      PDF [a.u.] : 0.07260124299183655
Energy[Mev]: 961.9238476954017      PDF [a.u.] : 0

nowy obraz mapy bitowej

I am wondering if there is some mistake or some conversion problem near 960 MeV beacuse for Landau distributions we get plain function with limit going to zero.

**Comment by marwin1991** [comment]
nowy obraz mapy bitowej

File: src/static/json/PhysicsRoutines/EnergyLossDistributions.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      "parameterName": "energy_loss_model",
+      "label": "Model",
+      "values": "energy_loss_distributions",
+      "defaultValue": 1,
+      "description": "Choose energy loss model"
+    },
+    {
+      "type": "input",
+      "parameterName": "E_MeV_u",
+      "label": "Energy [MeV]",
```
[the remark this answers is not in the document]
Comment (grzanka) [reply src/static/json/PhysicsRoutines/EnergyLossDistributions.json]: Funny, same happens to me

File: src/functionsFromC/AT_energy_loss_distributions.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    /*********************OUTPUT ARRAY*******************************/
+    let fDdDReturnData = new Float64Array(new Array(n));
+    let fDdDReturnDataBytesNumber = fDdDReturnData.length * fDdDReturnData.BYTES_PER_ELEMENT;
+    let fDdDReturnDataPointer = Module._malloc(fDdDReturnDataBytesNumber);
+    let fDdDReturnHeap = new Uint8Array(Module.HEAPF64.buffer, fDdDReturnDataPointer, fDdDReturnDataBytesNumber);
+
+    /*********************CALL FUNCTION******************************/
+
+    if( energy_loss_model == 1 ){ // Vavilov
```
[the remark this answers is not in the document]
Comment (grzanka) [reply src/functionsFromC/AT_energy_loss_distributions.js]: OK, I'll fix

**Comment by grzanka** [comment]
> For default values and Vavilov we have enrgy
> Energy[Mev] : 386.77354709418773 PDF [a.u.] : 0
> Energy[Mev] : 388.7775551102198 PDF [a.u.] : 0.020241360601070967
> .
> .
> .
> Energy[Mev]: 959.9198396793696 PDF [a.u.] : 0.07260124299183655
> Energy[Mev]: 961.9238476954017 PDF [a.u.] : 0
> 
> nowy obraz mapy bitowej
> 
> I am wondering if there is some mistake or some conversion problem near 960 MeV beacuse for Landau distributions we get plain function with limit going to zero.

First of all: keV, not MeV. I suspect this is a feature of Vavilov distribution: it doesn't have infinitely long right tail. Check http://vlado.fmf.uni-lj.si/texceh/kako/tex2pdf/ex/maria-1.pdf

> Vavilov[5] derived a more accurate straggling distribution by introducing the kinematic
> limit on the maximum transferable energy in a single collision, rather than using E max =
> ∞.

File: src/static/json/PhysicsRoutines/EnergyLossDistributions.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (grzanka) [reply src/static/json/PhysicsRoutines/EnergyLossDistributions.json]: Moved to #184. Simple setting of autofill to false doesn't help :(

**Comment by grzanka** [comment]
@marwin1991 please review again

**Comment by marwin1991** [comment]
I will investigate this card issue tomorrow. Maybe some class name or something makes browser to recognise this field as card number.

**Pull request merged**